api/cmd/portainer/main.go

```diff
@@ -171,6 +171,10 @@ func retrieveFirstEndpointFromDatabase(endpointService portainer.EndpointService
 func main() {
 	flags := initCLI()
 
+	if *flags.CheckHealth {
+		http.GetHealth("localhost" + *flags.Addr)
```

Comment:
See my comment below to manage the `os.Exit` part here (use `log.Fatal` in case of error).